src/loader.js

```diff
@@ -284,6 +284,48 @@ function setupModuleLoader(window) {
 
           /**
            * @ngdoc method
+           * @name angular.Module#component
+           * @module ng
+           * @param {string} name Component name
+           * @param {Object} options Component definition object
+           * @description
+           * TODO...
+           */
+          component: function(name, options) {
+            function factory($injector) {
+              function makeInjectable(fn) {
+                if (angular.isFunction(fn)) {
+                  return function(tElement, tAttrs) {
+                    return $injector.invoke(fn, this, {$element: tElement, $attrs: tAttrs});
+                  };
```

Comment:
I'm not sure how that would work, the methods already have a closure when they are placed on the options obj, I don't think we can update that closure. Best we could do is pass the dependencies as a map argument to the methods, but I think it really isn't worth it, it doesn't seem like a big performance hit for me.

Another option is to have module.component accept either a options object or a factory method and then if you want templateUrl/template function that use DI, you have to pass a factory instead of options. But again, I think having templateUrl/template functions injectable is much more elegant and not such a big perf hit. It is also consistent with how the controller and $canActivate will be invoked.